llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/Interval.h

```diff
@@ -127,6 +127,12 @@ template <typename T> class Interval {
   }
   /// Inequality.
   bool operator!=(const Interval &Other) const { return !(*this == Other); }
+  /// \Returns true if this interval comes before \p Other in program order.
+  /// This expects disjoint intervals.
+  bool comesBefore(const Interval &Other) const {
+    assert(disjoint(Other) && "Expect disjoint intervals!");
+    return bottom()->comesBefore(Other.top());
+  }
```

Comment:
Interval defines both {top, bottom} and {From, To} which are the same thing.  Should we just drop one?  Why not just have only {From, To} and from() and to().

Reply:
I prefer top/bottom because it implies program order. From/To could be both top->down or bottom->up.
I will upload a separate NFC patch to switch from From/To to Top/Bottom. I

Reply:
LGTM